src/renderer/redux/actions/subscriptions.js

```diff
@@ -14,5 +21,98 @@ export const doChannelUnsubscribe = (subscription: Subscription) => (dispatch: D
     data: subscription,
   });
 
+export const doCheckSubscriptions = () => (
+  dispatch: Dispatch,
+  getState: () => SubscriptionState
+) => {
+  const checkSubscriptionsTimer = setInterval(
+    () =>
+      selectSubscriptions(getState()).map((subscription: Subscription) =>
+        dispatch(doCheckSubscription(subscription))
+      ),
+    CHECK_SUBSCRIPTIONS_INTERVAL
+  );
+  dispatch({
+    type: ACTIONS.CHECK_SUBSCRIPTIONS_SUBSCRIBE,
+    data: { checkSubscriptionsTimer },
+  });
+};
+
+export const doCheckSubscription = (subscription: Subscription) => (dispatch: Dispatch) => {
+  dispatch({
+    type: ACTIONS.CHECK_SUBSCRIPTION_STARTED,
+    data: subscription,
+  });
+
+  Lbry.claim_list_by_channel({ uri: subscription.uri, page: 1 }).then(result => {
+    const claimResult = result[subscription.uri] || {};
+    const { claims_in_channel: claimsInChannel } = claimResult;
+
+    let count = subscription.latest
+      ? claimsInChannel.reduce(
+          (prev, cur, index) =>
+            buildURI({ contentName: cur.name, claimId: cur.claim_id}, false) === subscription.latest ? index : prev,
+          -1
+        )
+      : 1;
+
+    if (count !== 0) {
+      if (!claimsInChannel[0].value.stream.metadata.fee) {
+        dispatch(
+          doPurchaseUri(buildURI({ contentName: claimsInChannel[0].name, claimId: claimsInChannel[0].claim_id }, false), { cost: 0 })
+        );
+      }
+
+      const notif = new window.Notification(subscription.channelName, {
+        body: `Posted ${claimsInChannel[0].value.stream.metadata.title}${
+          count > 1 ? ` and ${count - 1} other new items` : ''
+        }${count < 0 ? ' and 9+ other new items' : ''}`,
+        silent: false,
+      });
+      notif.onclick = () => {
+        dispatch(
+          doNavigate('/show', {
+            uri: buildURI({ contentName: claimsInChannel[0].name, claimId: claimsInChannel[0].claim_id }, true),
+          })
+        );
+      };
+    }
+
+    //$FlowIssue
+    dispatch({
+      type: ACTIONS.CHECK_SUBSCRIPTION_COMPLETED,
+      data: subscription,
+    });
+  });
+};
+
+export const checkSubscriptionLatest = (channel: Subscription, uri: string) => (
```

Comment:
Is this function actually necessary? Could `doCheckSubscription` just call `setSubscriptionLatest`?

Reply:
check and set are different, set simply sets the value in the redux store. Check does an asynchronous call and sets the value accordingly by dispatching set, if that makes sense.

Reply:
I'm asking two things in my question:
1) If we're already fetching the latest subscription data at time interval `n`, is it actually necessary to check for the latest for this same information when accessing a file page?
2) If it is necessary (for staleness reasons, presumably), why can't more of the code be shared between the two? Isn't `doCheckSubscription` a superset of the functionality provided by `checkSubscriptionLatest`? and why wouldn't you want `checkSubscriptionLatest` to also begin downloads? Furthermore, why do _both_ of these call `Lbry.claim_list_by_channel` when they could call `doFetchClaimsByChannel` instead?

Reply:
@liamcardenas did you see above?